### PR TITLE
terraform-backend-git 0.0.13 (new formula)

### PR DESCRIPTION
**Pull request by dee-kryvenko**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/terraform-backend-git.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  homepage "https://github.com/plumber-cd/terraform-backend-git"
+  url "https://github.com/plumber-cd/terraform-backend-git/archive/v0.0.13.tar.gz"
+  sha256 "a97c4b4c56cff393550d5da1ae4add9e37e353e569d5846f680506e88a44027c"
+
+  bottle do
+    cellar :any_skip_relocation
+    # sha256 "a032752d893a2be4c7ade05fed422146c879a5cd4bcf69c8bee4489050ee7244" => :catalina
+    # sha256 "44d9ee3cd35d32a1dc33fde30c3e250bd0ac8c92ad6b99451dc2df3c785f59dd" => :mojave
+    # sha256 "21588d35f03ccd744d7c4cea52c5e44e34f6a8830d23710bf95965f04280cdf5" => :high_sierra
+  end
```
Comment (chenrui333) [inline Formula/terraform-backend-git.rb:12]: no need to have bottle block for new formula

Comment (chenrui333) [inline Formula/terraform-backend-git.rb:12]:
```suggestion
```

File: Formula/terraform-backend-git.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  depends_on "go" => :build
+  depends_on "terraform"
+
+  def install
+    system "go", "build", "-ldflags", "-X main.VERSION=v#{version}", *std_go_args
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/terraform-backend-git version")
```
Comment (chenrui333) [inline Formula/terraform-backend-git.rb:15]: version test is not enough, better to have sort of real test. checkout how git tools been tested.

**Comment by dee-kryvenko** [comment]
@chenrui333 I think I fixed all tests failures and accounted for your feedback. Seems like it's failing due to the repository is "not notable enough".

I understand why there is such a requirement and if you refuse to merge, I'll understand.

However, please consider the following - this Terraform HTTP backend implementation meant to be a part of hashicorp/terraform (notable enough, I think), as backends being a part of the core code base is the only options atm. Scalability issue with this approach is well understood by now, and they are going to make backends a pluggable bits https://github.com/hashicorp/terraform/issues/5877. When it's done, backends will be discoverable and consumable by internal Terraform mechanisms, so `terraform-backend-git` will not need to be distributed as a separate binary via brew or else. Therefore, this formula (if accepted) will be having limited lifetime and expected to be deprecated at some point.

However, getting there might take quite a some time, and meanwhile I was hoping brew could help giving this project a spin in terms of discoverability and lowering the entrance barrier. Is there a chance for it?

**Comment by chenrui333** [comment]
Hey @dee-kryvenko and thanks for the PR!

While this seems like pretty cool software, the audit failures mean it's not notable enough for us to package in Homebrew for now. This is not a decision on the merits of the software, but we cannot distribute every single piece of software and have to draw a line on "notability" somewhere. Of course, we would be happy to reconsider in the future.

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Comment by chenrui333** [comment]
I am personally a terraform user and a SRE, will definitely try out your git backend (which seems very cool).

**Pull request closed without merging**